src/nodes.coffee

```diff
@@ -966,6 +966,22 @@ exports.RegexLiteral = class RegexLiteral extends Literal
     val = replaceUnicodeCodePointEscapes val, {@flags}
     @value = "#{makeDelimitedLiteral val, delimiter: '/'}#{@flags}"
 
+  REGEX_REGEX: /// ^ / (.*) / \w* $ ///
+
+  astType: -> 'RegExpLiteral'
+
+  astProperties: ->
```

Comment:
Here we're going for semantic compatibility with Babel (the AST generated for a regex/non-interpolated heregex should match the AST Babel would generate for the corresponding JS-compiled regex)

Then we extend the Babel type with additional properties (`originalPattern`, `extra.originalRaw`) that preserve the original source formatting (eg whitespace in heregex, unicode code point escapes in a non-`u` regex)